### PR TITLE
add support for delta aggregation to otlp metrics

### PR DESCRIPTION
**Pull request by garypen**
Add a new configuration option (Temporality) to the otlp metrics configuration.

This may be useful to fix problems with metrics when being processed by datadog which tends to expect Delta, rather than Cumulative, aggregations.

See:
 - https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/6129
 - https://github.com/DataDog/documentation/pull/15840

for more details.

**Checklist**

Complete the checklist (and note appropriate exceptions) before a final PR is raised.

- [x] Changes are compatible[^1]
- [x] Documentation[^2] completed
- [x] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [ ] Unit Tests
    - [ ] Integration Tests
    - [ ] Manual Tests

**Exceptions**

*Note any exceptions here*

**Notes**

[^1]. It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
[^2]. Configuration is an important part of many changes. Where applicable please try to document configuration examples.
[^3]. Tick whichever testing boxes are applicable. If you are adding Manual Tests:
    - please document the manual testing (extensively) in the Exceptions.
    - please raise a separate issue to automate the test and label it (or ask for it to be labeled) as `manual test`

### CODE REVIEW

File: apollo-router/src/plugins/telemetry/otlp.rs [hunk 1]
```diff
@@ -45,6 +45,10 @@ pub(crate) struct Config {
     /// Batch processor settings
     #[serde(default)]
     pub(crate) batch_processor: BatchProcessorConfig,
+
+    /// Temporality
```
Comment (BrynCooke) [inline apollo-router/src/plugins/telemetry/otlp.rs:49]:
```suggestion
    /// Temporality for export (default: `Cumulative`).
    /// Note that when exporting to Datadog agent use `Delta`.
```

File: apollo-router/src/plugins/telemetry/otlp.rs [hunk 2]
```diff
@@ -203,6 +207,14 @@ pub(crate) enum Protocol {
     Http,
 }
 
+#[derive(Debug, Default, Clone, Deserialize, Serialize, JsonSchema)]
+#[serde(deny_unknown_fields, rename_all = "snake_case")]
+pub(crate) enum Temporality {
+    #[default]
+    Cumulative,
+    Delta,
```
Comment (BrynCooke) [inline apollo-router/src/plugins/telemetry/otlp.rs:215]:
```suggestion
    /// Export cumulative metrics.
    #[default]
    Cumulative,
    /// Export delta metrics. `Delta` should be used when exporting to DataDog Agent.
    Delta,
```

**Review by BrynCooke — APPROVED** [review]

**Review by o0Ignition0o — APPROVED** [review]

**Comment by bnjjj** [comment]
@garypen maybe as a follow up we should add more documentation about this in the right section of our docs ?

**Comment by garypen** [comment]
> @garypen maybe as a follow up we should add more documentation about this in the right section of our docs ?

That is the plan. We are making sure we understand the impact before doing that. I'll file an issue.

**Pull request merged**